wp-includes/sqlite/class-wp-sqlite-translator.php

```diff
@@ -3755,7 +3756,25 @@ private function get_key_definitions( $table_name, $columns ) {
 			$key_definition[] = sprintf( '`%s`', $index_name );
 
 			$cols = array_map(
-				function ( $column ) {
+				function ( $column ) use ( $table_name, $key_length_limit ) {
+					$data_type   = strtolower( $this->get_cached_mysql_data_type( $table_name, $column['name'] ) );
+					$data_length = $key_length_limit;
+
+					// Extract the length from the data type. Make it lower if needed. Skip 'unsigned' parts and whitespace.
+					if ( 1 === preg_match( '/^(\w+)\s*\(\s*(\d+)\s*\)\s*(.*)?$/', $data_type, $matches ) ) {
```

Comment:
This is just a tiny nitpick:
- I guess the trailing `(.*)?` is equivalent to just `.*`, or we can simply drop the `$` as anchoring the beginning is enough.
- The `Skip 'unsigned' parts` piece of the comment is probably rather irrelevant, as we're not trying to match integers.

Reply:
Thanks for ideas, I cleaned up this regex more.